internal/controller/nonadmindownloadrequest_controller.go

```diff
@@ -0,0 +1,188 @@
+	client.Client
+	Scheme        *runtime.Scheme
+	OADPNamespace string
+}
+
+// +kubebuilder:rbac:groups=oadp.openshift.io,resources=nonadmindownloadrequests,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=oadp.openshift.io,resources=nonadmindownloadrequests/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=oadp.openshift.io,resources=nonadmindownloadrequests/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the NonAdminDownloadRequest object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.20.2/pkg/reconcile
+//
+// This reconcile implements ObjectReconciler https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.20.2/pkg/reconcile#ObjectReconciler
+// Each reconciliation event gets the associated object from Kubernetes before passing it to Reconcile
+func (r *NonAdminDownloadRequestReconciler) Reconcile(ctx context.Context, req *nacv1alpha1.NonAdminDownloadRequest) (reconcile.Result, error) {
+	if req == nil || req.Spec.Target.Kind == constant.EmptyString {
+		return ctrl.Result{Requeue: false}, nil
+	}
+	// TODO: delete if UID is always available here.
+	// if req.ObjectMeta.UID  == "" {
+	// requeue later until uid is populated
+	// }
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling NonAdminDownloadRequest", "name", req.Name, "namespace", req.Namespace)
+	if req.Status.VeleroDownloadRequestStatus.Expiration.After(time.Now()) {
+		// request is expired, so delete NADR after deleting velero DR
+		// find associated downloadrequest and delete that first
+		logger.V(1).Info("Deleting expired NonAdminDownloadRequest", req.Name, req.Namespace)
+		if err := r.Delete(ctx, req); err != nil {
+			if apierrors.IsNotFound(err) {
+				return reconcile.Result{Requeue: false}, nil
+			}
+			return reconcile.Result{Requeue: true}, nil
+		}
+	}
+	var veleroObj, naObj client.Object
+	switch req.Spec.Target.Kind {
+	case velerov1.DownloadTargetKindBackupLog,
+		velerov1.DownloadTargetKindBackupContents,
+		velerov1.DownloadTargetKindBackupVolumeSnapshots,
+		velerov1.DownloadTargetKindBackupItemOperations,
+		velerov1.DownloadTargetKindBackupResourceList,
+		velerov1.DownloadTargetKindBackupResults,
+		velerov1.DownloadTargetKindCSIBackupVolumeSnapshots,
+		velerov1.DownloadTargetKindCSIBackupVolumeSnapshotContents,
+		velerov1.DownloadTargetKindBackupVolumeInfos:
+		// get velero backup name from nab
+		var nab nacv1alpha1.NonAdminBackup
+		if err := r.Get(ctx, types.NamespacedName{Namespace: r.OADPNamespace, Name: req.Spec.Target.Name}, &nab); err != nil {
```

Comment:
Shouldn't Namespace here be `req.Namespace`? The NAB will be in the same namespace as the NADR, not the velero/OADP namespace.